recommendation-model/utils/dataset.py

```diff
@@ -0,0 +1,234 @@
+    DEFAULT_ITEM_CATEGORY,
+    DEFAULT_RETAIL_PRICE,
+    MERCHANT_PRODUCT_PRICE,
+    DISCOUNTED_PRICE,
+    MERCHANT_CATEGORY_ID,
+    PRICE_LEVEL
+)
+
+
+
+def load_pandas_df(
+    header=None):
+    """
+    Loads the Mongodb  dataset as pd.DataFrame.
+    To load customer information only, you can use load_item_df function.
+
+    Args:
+        size (str): Size of the data to load. One of ("100k", "1m", "10m", "20m").
+        header (list or tuple or None): Rating dataset header.
+        local_cache_path (str): Path (directory or a zip file) to cache the downloaded zip file.
+            If None, all the intermediate files will be stored in a temporary directory and removed after use.
+        title_col (str): Movie title column name. If None, the column will not be loaded.
+        genres_col (str): Genres column name. Genres are '|' separated string.
+            If None, the column will not be loaded.
+        year_col (str): Movie release year column name. If None, the column will not be loaded.
+
+    Returns:
+        pd.DataFrame: Movie rating dataset.
+        
+
+    **Examples**
+
+    .. code-block:: python
+    
+        # To load just user-id, item-id, and ratings from MovieLens-1M dataset,
+        df = load_pandas_df('1m', ('UserId', 'ItemId', 'Rating'))
+
+        # To load rating's timestamp together,
+        df = load_pandas_df('1m', ('UserId', 'ItemId', 'Rating', 'Timestamp'))
+
+        # To load movie's title, genres, and released year info along with the ratings data,
+        df = load_pandas_df('1m', ('UserId', 'ItemId', 'Rating', 'Timestamp'),
+            title_col='Title',
+            genres_col='Genres',
+            year_col='Year'
+        )
+    """
+    if header is None:
+        header = DEFAULT_HEADER
+    elif len(header) < 2:
+        raise ValueError(ERROR_HEADER)
+    elif len(header) > 4:
+        warnings.warn(WARNING_FCGL_HEADER)
+        # header = header[:4]
+
+    fcgl_column = header[1]
+    # collection_name : Collection name that is used to train the vowpal wabbit model.
+    # collection_name : Since this is what helps in recommnding we will push in the intial user data we get from the user while he/she
+    # signups for the application.
+
+
+    df = DataFrame( list(mydb.training_data.find({})) )
+    
+    return df
+
+
+def load_item_df(
+    movie_col=DEFAULT_ITEM_COL,
+    title_col=None,
+    genres_col=None,
+    year_col=None):
+    """Loads Movie info.
+
+    Args:
+        movie_col (str): Movie id column name.
+        title_col (str): Movie title column name. If None, the column will not be loaded.
+        genres_col (str): Genres column name. Genres are '|' separated string.
+            If None, the column will not be loaded.
+        year_col (str): Movie release year column name. If None, the column will not be loaded.
+
+    Returns:
+        pd.DataFrame: Movie information data, such as title, genres, and release year.
+    """
+    size = size.lower()
```

Comment:
Not sure where some of these variables are coming from. I don't see `size` anywhere

Reply:
Oh I think it was supposed to be one of the function params. It just wasn't added in yet load_item_df(`size`, ...)